lib/oauth.js

```diff
@@ -167,9 +167,17 @@ exports.v2 = function (settings) {
             }
 
             query.client_id = settings.clientId;
-            query.response_type = 'code';
+            query.response_type = settings.provider.response_type;
             query.redirect_uri = internals.location(request, protocol, settings.location);
             query.state = nonce;
+            if (settings.provider.nonce) {
+                query.nonce = nonce; // required by microsoftonline
```

Comment:
Well, it looks like you're implementing `3.3.  Authentication using the Hybrid Flow`
instead of the more traditional `3.1.  Authentication using the Authorization Code Flow` that bell assumes. I am talking specifically about the OpenID Connect spec @ http://openid.net/specs/openid-connect-core-1_0.html

In that case the nonce is indeed required. Also, not using a different value could allow an attacker to more easily identify the key we use to encrypt the cookie we keep around to validate the state. That's definitely a big problem.